lib/cosmos/interfaces/protocols/protocol.rb

```diff
@@ -15,9 +15,12 @@
 module Cosmos
   class Protocol
     attr_accessor :interface
+    attr_accessor :allow_empty_data
 
-    def initialize
+    # @param allow_empty_data [true/false] Whether STOP should be returned on empty data
+    def initialize(allow_empty_data = false)
```

Comment:
I can see your point but if you don't have a concrete example of using this in our code I don't know how anyone else will know when to use it. :face_with_head_bandage: